### ISSUE TITLE
Fedora Copr repo issue

### ISSUE THREAD

**Issue by ljubacanoa** [Hjdskes/gcolor3#85]
For some reasons, after upgrade to F28, I have "Failed to synchronize cache for repo 'fnux-gcolor3', disabling.".
I removed it, than enabled via command line again and same message.

**Comment by Hjdskes** [comment]
@Fnux, can you fix this?

**Comment by Fnux** [comment]
I will take a look this afternoon (CET).

**Comment by Fnux** [comment]
The package failed to build for F28, using the tarball for version 2.2:

```
        compiler:               gcc
        cflags:                 -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -mcet -fcf-protection
        ldflags:                -Wl,-z,relro  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld

[...]

gcolor3-application.c: In function ‘gcolor3_application_action_quit’:
gcolor3-application.c:80:27: error: cast between incompatible function types from ‘void (*)(GtkWidget *)’ {aka ‘void (*)(struct _GtkWidget *)’} to ‘void (*)(void *, void *)’ [-Werror=cast-function-type]
  g_list_foreach (windows, (GFunc) gtk_widget_destroy, NULL);
                           ^
cc1: all warnings being treated as errors
```

@Hjdskes, do you see an obvious fix ? The compile flags are provided by the distribution and I don't want to change them. The ideal would be to include a patch in the package.

**Comment by Hjdskes** [comment]
I'll take a look at it soon. I suppose I will merge the Meson PR, fix this and release a new version that you can package for F28.

**Comment by Fnux** [comment]
@Hjdskes any progress on the new release ? I guess I can take a look next week but I never played with GTK before.

**Comment by Hjdskes** [comment]
I'm sorry, I really do not have much time to work on Gcolor3 at the moment. Regarding the error, you can try a wrapper function `void destroy(GtkWidget *widget, gpointer user_data)` which calls `gtk_widget_destroy(widget)`, i.e.

```
void destroy(GtkWidget *widget, gpointer user_data)
{
  gtk_widget_destroy(widget);
}
```

**Comment by Fnux** [comment]
I rebuilt the package with a simple patch: https://copr.fedorainfracloud.org/coprs/fnux/gcolor3/build/771049/

You can now sync the repo on F28/rawhide(=F29).

**Comment by Fnux** [comment]
@Hjdskes, you can close the issue.

**Comment by Hjdskes** [comment]
I'll keep it open as a reminder to include the workaround.

**Comment by Hjdskes** [comment]
Fixed in 1f84777, closing.
